chasm/lib/scheduler/invoker_tasks.go

```diff
@@ -0,0 +1,610 @@
+) (*schedulepb.ScheduleActionResult, error) {
+	requestSpec := scheduler.GetSchedule().GetAction().GetStartWorkflow()
+	nominalTimeSec := start.NominalTime.AsTime().Truncate(time.Second)
+	workflowID := fmt.Sprintf("%s-%s", requestSpec.WorkflowId, nominalTimeSec.Format(time.RFC3339))
+
+	if start.Attempt >= InvokerMaxStartAttempts {
+		return nil, errRetryLimitExceeded
+	}
+
+	// Get rate limiter permission once per buffered start, on the first attempt only.
+	if start.Attempt == 1 {
+		delay, err := e.getRateLimiterPermission()
+		if err != nil {
+			return nil, err
+		}
+		if delay > 0 {
+			return nil, newRateLimitedError(delay)
+		}
+	}
+
+	reusePolicy := enumspb.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE
+	if start.Manual {
+		reusePolicy = enumspb.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE
+	}
+
+	// TODO - set last completion result/continued failure (watcher)
+	// TODO - set search attributes
+	request := &workflowservice.StartWorkflowExecutionRequest{
+		Namespace:                scheduler.Namespace,
+		WorkflowId:               workflowID,
+		WorkflowType:             requestSpec.WorkflowType,
+		TaskQueue:                requestSpec.TaskQueue,
+		Input:                    requestSpec.Input,
+		WorkflowExecutionTimeout: requestSpec.WorkflowExecutionTimeout,
+		WorkflowRunTimeout:       requestSpec.WorkflowRunTimeout,
+		WorkflowTaskTimeout:      requestSpec.WorkflowTaskTimeout,
+		Identity:                 scheduler.identity(),
+		RequestId:                start.RequestId,
+		WorkflowIdReusePolicy:    reusePolicy,
+		RetryPolicy:              requestSpec.RetryPolicy,
+		Memo:                     requestSpec.Memo,
+		SearchAttributes:         nil,
+		Header:                   requestSpec.Header,
+		LastCompletionResult:     nil,
+		ContinuedFailure:         nil,
+		UserMetadata:             requestSpec.UserMetadata,
+	}
+	result, err := e.FrontendClient.StartWorkflowExecution(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &schedulepb.ScheduleActionResult{
+		ScheduleTime: start.ActualTime,
+		ActualTime:   timestamppb.New(time.Now()),
+		StartWorkflowResult: &commonpb.WorkflowExecution{
+			WorkflowId: workflowID,
+			RunId:      result.RunId,
+		},
+		StartWorkflowStatus: result.Status, // usually should be RUNNING
+	}, nil
+}
+
+func (e *InvokerExecuteTaskExecutor) terminateWorkflow(
+	ctx context.Context,
+	scheduler *Scheduler,
+	target *commonpb.WorkflowExecution,
+) error {
+	request := &historyservice.TerminateWorkflowExecutionRequest{
+		NamespaceId: scheduler.NamespaceId,
+		TerminateRequest: &workflowservice.TerminateWorkflowExecutionRequest{
+			Namespace:           scheduler.Namespace,
+			WorkflowExecution:   &commonpb.WorkflowExecution{WorkflowId: target.WorkflowId},
+			Reason:              "terminated by schedule overlap policy",
+			Identity:            scheduler.identity(),
+			FirstExecutionRunId: target.RunId,
+		},
+	}
+	_, err := e.HistoryClient.TerminateWorkflowExecution(ctx, request)
```

Comment:
Why terminate & cancel goes through history client instead frontend like start workflow does? (related to Action counting?)

Reply:
Can you put a comment in the code explaining this detail?
Later on we should discuss if we can enforce metering in the history service directly, not for this PR.